src-self-hosted/link.zig

```diff
@@ -560,25 +607,26 @@ pub const File = struct {
 
         pub fn populateMissingMetadata(self: *Elf) !void {
             const small_ptr = switch (self.ptr_width) {
-                .p32 => true,
+                .p16, .p32 => true,
                 .p64 => false,
             };
```

Comment:
ELF doesn't have a 16 bit format. The changes here look kinda smelly. What do you think about having a separate struct for this ELF16 format (along with link.Elf and link.C)?